build derrière un proxy: Souci nosexcover

L'installation de nosexcover par le make ne passe pas. en utilisant un pip install manuel, il n'y a pas de souci par contre

```
[root@workshop3-local geoportailv3]# make -f jkonnen.mk build
mkdir -p .build/
virtualenv --no-site-packages .build/venv
New python executable in .build/venv/bin/python
Installing Setuptools..............................................................................................................................................................................................................................done.
Installing Pip.....................................................................................................................................................................................................................................................................................................................................done.
touch .build/venv.timestamp
.build/venv/bin/pip install -r requirements.txt
Obtaining file:///tmp/geoportailv3 (from -r requirements.txt (line 4))
  Running setup.py egg_info for package from file:///tmp/geoportailv3
    /usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'message_extractors'
      warnings.warn(msg)

Obtaining pyramid-closure from git+https://github.com/camptocamp/pyramid_closure#egg=pyramid_closure (from -r requirements.txt (line 3))
  Cloning https://github.com/camptocamp/pyramid_closure to ./.build/venv/src/pyramid-closure
  Running setup.py egg_info for package pyramid-closure

    warning: no files found matching '*.ini'
    warning: no files found matching '*.cfg'
    warning: no files found matching '*.rst'
    warning: no files found matching '*.ico' under directory 'pyramid_closure'
    warning: no files found matching '*.png' under directory 'pyramid_closure'
    warning: no files found matching '*.css' under directory 'pyramid_closure'
    warning: no files found matching '*.gif' under directory 'pyramid_closure'
    warning: no files found matching '*.jpg' under directory 'pyramid_closure'
    warning: no files found matching '*.pt' under directory 'pyramid_closure'
    warning: no files found matching '*.mak' under directory 'pyramid_closure'
    warning: no files found matching '*.mako' under directory 'pyramid_closure'
    warning: no files found matching '*.html' under directory 'pyramid_closure'
    warning: no files found matching '*.xml' under directory 'pyramid_closure'
Downloading/unpacking c2cgeoportal>=1.6.0dev-20141107 (from geoportailv3==1.0->-r requirements.txt (line 4))
  Downloading c2cgeoportal-1.6.0dev-20141121.tar.gz (34.2MB): 34.2MB downloaded
  Running setup.py egg_info for package c2cgeoportal
    Not found: ./c2cgeoportal/tests/testegg/dist/testegg-1.0.tar.gz
    Download error on https://pypi.python.org/simple/nosexcover/: [Errno 1] _ssl.c:504: error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol -- Some packages may not be found!
   ** Couldn't find index page for 'nosexcover' (maybe misspelled?)
    Download error on https://pypi.python.org/simple/: [Errno 1] _ssl.c:504: error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol -- Some packages may not be found!
    No local packages or download links found for nosexcover==1.0.8 **
    Traceback (most recent call last):
      File "<string>", line 16, in <module>
      File "/tmp/geoportailv3/.build/venv/build/c2cgeoportal/setup.py", line 145, in <module>
        'admin = c2cgeoportal.forms:fanstatic_lib',
      File "/usr/lib64/python2.7/distutils/core.py", line 112, in setup
        _setup_distribution = dist = klass(attrs)
      File "/tmp/geoportailv3/.build/venv/lib/python2.7/site-packages/setuptools/dist.py", line 265, in __init__
        self.fetch_build_eggs(attrs.pop('setup_requires'))
      File "/tmp/geoportailv3/.build/venv/lib/python2.7/site-packages/setuptools/dist.py", line 289, in fetch_build_eggs
        parse_requirements(requires), installer=self.fetch_build_egg
      File "/tmp/geoportailv3/.build/venv/lib/python2.7/site-packages/pkg_resources.py", line 618, in resolve
        dist = best[req.key] = env.best_match(req, self, installer)
      File "/tmp/geoportailv3/.build/venv/lib/python2.7/site-packages/pkg_resources.py", line 862, in best_match
        return self.obtain(req, installer) # try and download/install
      File "/tmp/geoportailv3/.build/venv/lib/python2.7/site-packages/pkg_resources.py", line 874, in obtain
        return installer(requirement)
      File "/tmp/geoportailv3/.build/venv/lib/python2.7/site-packages/setuptools/dist.py", line 339, in fetch_build_egg
        return cmd.easy_install(req)
      File "/tmp/geoportailv3/.build/venv/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 617, in easy_install
        raise DistutilsError(msg)
    distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('nosexcover==1.0.8')
    Complete output from command python setup.py egg_info:
    Not found: ./c2cgeoportal/tests/testegg/dist/testegg-1.0.tar.gz

Download error on https://pypi.python.org/simple/nosexcover/: [Errno 1] _ssl.c:504: error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol -- Some packages may not be found!

Couldn't find index page for 'nosexcover' (maybe misspelled?)

Download error on https://pypi.python.org/simple/: [Errno 1] _ssl.c:504: error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol -- Some packages may not be found!

No local packages or download links found for nosexcover==1.0.8

Traceback (most recent call last):

  File "<string>", line 16, in <module>

  File "/tmp/geoportailv3/.build/venv/build/c2cgeoportal/setup.py", line 145, in <module>

    'admin = c2cgeoportal.forms:fanstatic_lib',

  File "/usr/lib64/python2.7/distutils/core.py", line 112, in setup

    _setup_distribution = dist = klass(attrs)

  File "/tmp/geoportailv3/.build/venv/lib/python2.7/site-packages/setuptools/dist.py", line 265, in __init__

    self.fetch_build_eggs(attrs.pop('setup_requires'))

  File "/tmp/geoportailv3/.build/venv/lib/python2.7/site-packages/setuptools/dist.py", line 289, in fetch_build_eggs

    parse_requirements(requires), installer=self.fetch_build_egg

  File "/tmp/geoportailv3/.build/venv/lib/python2.7/site-packages/pkg_resources.py", line 618, in resolve

    dist = best[req.key] = env.best_match(req, self, installer)

  File "/tmp/geoportailv3/.build/venv/lib/python2.7/site-packages/pkg_resources.py", line 862, in best_match

    return self.obtain(req, installer) # try and download/install

  File "/tmp/geoportailv3/.build/venv/lib/python2.7/site-packages/pkg_resources.py", line 874, in obtain

    return installer(requirement)

  File "/tmp/geoportailv3/.build/venv/lib/python2.7/site-packages/setuptools/dist.py", line 339, in fetch_build_egg

    return cmd.easy_install(req)

  File "/tmp/geoportailv3/.build/venv/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 617, in easy_install

    raise DistutilsError(msg)

distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('nosexcover==1.0.8')

----------------------------------------
Cleaning up...
Command python setup.py egg_info failed with error code 1 in /tmp/geoportailv3/.build/venv/build/c2cgeoportal
Storing complete log in /root/.pip/pip.log
make: *** [/geoportailv3.egg-link] Error 1
[root@workshop3-local geoportailv3]# http://jasper.geoportal.lu:8080/service?REQUEST=GetCapabilities^C
**[root@workshop3-local geoportailv3]# source .build/venv/bin/activate**
**(venv)[root@workshop3-local geoportailv3]# pip install nosexcover==1.0.8**
Downloading/unpacking nosexcover==1.0.8
  Downloading nosexcover-1.0.8.tar.gz
  Running setup.py egg_info for package nosexcover

Downloading/unpacking nose (from nosexcover==1.0.8)
  Downloading nose-1.3.4.tar.gz (277kB): 277kB downloaded
  Running setup.py egg_info for package nose

    no previously-included directories found matching 'doc/.build'
Downloading/unpacking coverage>=3.4 (from nosexcover==1.0.8)
  Downloading coverage-3.7.1.tar.gz (284kB): 284kB downloaded
  Running setup.py egg_info for package coverage

    warning: no previously-included files matching '*.pyc' found anywhere in distribution
Installing collected packages: nosexcover, nose, coverage
  Running setup.py install for nosexcover

  Running setup.py install for nose

    no previously-included directories found matching 'doc/.build'
    Installing nosetests script to /tmp/geoportailv3/.build/venv/bin
    Installing nosetests-2.7 script to /tmp/geoportailv3/.build/venv/bin
  Running setup.py install for coverage
    building 'coverage.tracer' extension
    gcc -pthread -fno-strict-aliasing -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include/python2.7 -c coverage/tracer.c -o build/temp.linux-x86_64-2.7/coverage/tracer.o
    gcc -pthread -shared -Wl,-z,relro build/temp.linux-x86_64-2.7/coverage/tracer.o -L/usr/lib64 -lpython2.7 -o build/lib.linux-x86_64-2.7/coverage/tracer.so

    warning: no previously-included files matching '*.pyc' found anywhere in distribution
    Installing coverage2 script to /tmp/geoportailv3/.build/venv/bin
    Installing coverage-2.7 script to /tmp/geoportailv3/.build/venv/bin
    Installing coverage script to /tmp/geoportailv3/.build/venv/bin
**Successfully installed nosexcover nose coverage**
Cleaning up...
```

--- a/CONST_Makefile
+++ b/CONST_Makefile
@@ -1,4 +1,5 @@
 SITE_PACKAGES = $(shell .build/venv/bin/python -c "import distutils; print(distutils.sysconfig.get_python_lib())" 2> /dev/null)
+PYTHON_VERSION = $(shell .build/venv/bin/python -c "import sys; print(sys.version_info.major + '.' + sys.version_info.minor)" 2> /dev/null)
 OL_JS_FILES = $(shell find node_modules/openlayers/src/ol -type f -name '*.js' 2> /dev/null)
 NGEO_JS_FILES = $(shell find node_modules/ngeo/src -type f -name '*.js' 2> /dev/null)
 APP_JS_FILES = $(shell find geoportailv3/static/js -type f -name '*.js')
@@ -168,5 +169,16 @@ geoportailv3/static/build/locale/fr/geoportailv3.json: geoportailv3/locale/fr/LC
 	virtualenv --no-site-packages .build/venv
 	touch $@
 
-$(SITE_PACKAGES)/geoportailv3.egg-link: .build/venv.timestamp setup.py requirements.txt
+$(SITE_PACKAGES)/geoportailv3.egg-link: .build/venv.timestamp setup.py \
+		requirements.txt \
+		$(SITE_PACKAGES)/nosexcover-1.0.8-py$(PYTHON_VERSION).egg-info/PKG-INFO \
+		$(SITE_PACKAGES)/setuptools_git-1.1-py$(PYTHON_VERSION).egg-info/PKG-INFO
 	.build/venv/bin/pip install -r requirements.txt
+
+# We manually install nosexcover and setuptools_git because of
+# https://github.com/Geoportail-Luxembourg/geoportailv3/issues/83
+$(SITE_PACKAGES)/nosexcover-1.0.8-py$(PYTHON_VERSION).egg-info/PKG-INFO:
+	.build/venv/bin/pip install "nosexcover==1.0.8"
+
+$(SITE_PACKAGES)/setuptools_git-1.1-py$(PYTHON_VERSION).egg-info/PKG-INFO:
+	.build/venv/bin/pip install "setuptools_git==1.1"

